cmd/controller/processevent.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/pkg/udev"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//processDiskFromUdevEvent process different event from udev-event module
+func processDiskFromUdevEvent(c *Controller, msg map[string]string) {
```

Comment:
Error case handle in parent function so no need to return error here.

Reply:
Thinking further, we need to make use of methods than functions. There is a code smell if functions has custom instances as it arguments. In other words, these custom instances e.g. *Controller should expose more methods i.e.
```go
// NOTE:
// Controller should be composed of a Device interface.
// One such Device interface can be UDevDevice.

func (c *Controller) processEvent() {...}
```
Above can also be a interface method depending on requirements e.g.

```go
type EventProcesser interface {
  ProcessEvent()
}

func (c *Controller) ProcessEvent() {...}
```

Since this entire controller is for Device/Disks we need not mention have 'Device' or 'Disk' strings in the method or function names. However, this is not a binding rule.

Reply:
changed it.